### PR TITLE
Refactor completion of generated document

### PR DESCRIPTION
**Pull request by jsimons1989**
Rename GeneratedDocument to DocumentGenerationCompleted as part of splitting submitted-data-document-generated into 3 separate topics (started, completed and failed).

Schema has been updated to have fields specific to the completion of the generated document:
- User ID
- GUID
- Description (temporarily hard coded for abridged accounts in generation service)
- Document size
- Document time (made up date for accounts)
- Document generation time

Changes have also been made to document-generator-accounts on the same feature branch name.

Resolves: FAPI-417, FAPI-418, FAPI-411

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/document/render/DocumentRenderRequestHandler.java [hunk 1]
```diff
@@ -17,29 +22,55 @@
...
+		String generatedDocumentJson;
+		
+		try(InputStream response = connection.getInputStream()) {
+			generatedDocumentJson = new String(IOUtils.toByteArray(response));
+		}
 		
-		generatedDocument.setStatus(connection.getResponseCode());
+		generatedDocument = RenderedDocumentJsonHandler.convert(generatedDocumentJson);
 		
 		if(connection.getResponseCode() == 201) {
```
Comment (davidmort) [inline src/main/java/uk/gov/companieshouse/document/render/DocumentRenderRequestHandler.java]: should use the HttpURLConnection static value HTTP_CREATED here, see [the documentation](https://docs.oracle.com/javase/7/docs/api/java/net/HttpURLConnection.html#HTTP_CREATED)

Comment (davidmort) [inline src/main/java/uk/gov/companieshouse/document/render/DocumentRenderRequestHandler.java]: use HTTP_OK [here](https://github.com/companieshouse/document-generator-core/blob/feature/document-generation-completed/src/main/java/uk/gov/companieshouse/avro/AvroSchemaHelper.java#L27) and [here](https://github.com/companieshouse/document-generator-core/blob/feature/document-generation-completed/src/main/java/uk/gov/companieshouse/document/data/DocumentDataHandler.java#L26)

**Comment by davidmort** [comment]
should the kafka config for retries be configurable? https://github.com/companieshouse/document-generator-core/blob/feature/document-generation-completed/src/main/java/uk/gov/companieshouse/kafka/DocumentGeneratorProducer.java#L26

**Comment by davidmort** [comment]
theres a test folder within the "main" folder src/main/test/uk/gov/companieshouse/accounts/AvroSerializerTest.java, is this right?

**Comment by mattrout92** [comment]
The value in the go library is also 10 retries and we just copied this. This comment is also not part of the stories we committed to completing this sprint. This will need to be in a new story.

**Comment by davidmort** [comment]
creating new stories is fine. just because the retries was set like this in the go library doesn't mean it's correct

**Pull request merged**